### ISSUE TITLE
I can't say I'm enamoured by pipx.

### ISSUE THREAD

**Issue by JGFMK** [pypa/pipx#1547]
All this has done is make commands longer to type. I don't like the verbosity this has introduced.
I was happier with virtual environments.
It took a while to figure out where it was installing things - all wasting my time.
I've seen how you can run pipx with the dependencies metadata.
Sometimes however you have multiple versions of the method  from different sources;
So you end up doing individual imports in methods where they are needed.
However the metadata only works at the top of the module.

I find pipx a thorn in my side and I've spent 4+ hours today fixing stuff after a Mac OS/X Sequoia upgrade.a b
Seriously you guys haven't put enough thought into this. It's like a pre beta product for me.

**Comment by mje-nz** [comment]
I don't understand what you're trying to achieve here exactly, and I suspect the devs didn't either and that's why you didn't get any responses. It sounds like you're trying to use pipx to manage your local development environments. Pipx is a tool to help you install and run end-user applications. I'm sure trying to use it to manage your local development environments is very uncomfortable, and you should not use it for that.

In the future, you may get better responses by following the bug report template, which will ensure your issue includes the information the devs need.

**Comment by gaby** [comment]
@JGFMK Agree, everytime I use `pipx` I have to type `--include-deps` and/or `--include-apps`. I don't understand why they are not enabled by default.

**Comment by MestreLion** [comment]
Why are you using `pipx` in the first place?

Pipx is a tool to download, install and run python _applications_ as if they were external commands. In a way, it's more akin to Snap and Flatpak than to `pip`.

It's not very useful (or at all) in an IDE like VSCode, as an IDE usually manages Virtual Environments for you,  activating them and setting the interpreter during development. And `pipx run somescript.py`  is not very useful when you want to _debug_ that script in an IDE. And it's not _that_ useful when you want to run your script inside a virtual environment you (or your IDE) created yourself. Just activate it and run it!

Also, you seem to misunderstood what the script metadata is for. It's a "mini-`pyproject.toml`" for single-module applications. It's _not_ a module import annotation! It's a _project dependency_ metadata when you _install_ a project.

For the other 2 sensible complains, about autocomplete in `zsh` and  `--include-deps` by default, I suggest creating distinct, focused issues.
